src/Voyager.php

```diff
@@ -68,12 +68,24 @@ public static function can($permission)
 
         if ($exist) {
             $user = User::find(Auth::id());
-            if ($user == null) {
+            if ($user == null || !$user->hasPermission($permission)) {
                 throw new UnauthorizedHttpException(null);
             }
-            if (!$user->hasPermission($permission)) {
-                throw new UnauthorizedHttpException(null);
+        }
+    }
+
+    public static function have($permission)
+    {
+        // Check if permission exist
+        $exist = Permission::where('key', $permission)->first();
+
+        if ($exist) {
+            $user = User::find(Auth::id());
```

Comment:
@marktopper I totally agree with those changes. We should also cache the permissions and flush them every time they are updated, like https://github.com/spatie/laravel-permission/blob/master/src/PermissionRegistrar.php#L70